src/Query/index.js

```diff
@@ -81,6 +88,14 @@ export default class Query<Record: Model> {
     return toPromise(callback => this.collection._fetchQuery(this, callback))
   }
 
+  then<U>(
```

Comment:
that's cool :)

Reply:
@kokusGr sadly, `await tenable` does not type-check in Flow right now :( there's a pending PR for it. but hopefully soon enough.